--- a/external/gpl3/gcc/lib/libstdc++-v3/Makefile
+++ b/external/gpl3/gcc/lib/libstdc++-v3/Makefile
@@ -84,6 +84,10 @@
 CPPFLAGS.thread.cc = -std=gnu++0x
 CPPFLAGS.future.cc = -std=gnu++0x
 
+# GCC13 moves these c++98 sources to C++11 semantics.
+CPPFLAGS.locale_init.cc += -std=gnu++11
+CPPFLAGS.localename.cc += -std=gnu++11
+
 CPPFLAGS.cp-demangle.cc += -DIN_GLIBCPP_V3
 CPPFLAGS.concept-inst.cc += -D_GLIBCXX_CONCEPT_CHECKS
 CPPFLAGS.parallel_list.cc += -D_GLIBCXX_PARALLEL
