--- a/external/gpl3/gcc/usr.bin/backend/Makefile
+++ b/external/gpl3/gcc/usr.bin/backend/Makefile
@@ -181,20 +181,21 @@
 			--host=`${HOST_SH} ${GNUHOSTDIST}/config.guess` \
 			--target=${MACHINE_GNU_PLATFORM} && \
 			mv auto-host.h ../auto-build.h) && \
 		rm -rf .ab
 .if defined(__MINIX)
 	${TOOL_SED} \
 		-e 's/^#define HAVE_MADVISE 1$$/#define HAVE_MADVISE 0/' \
 		-e 's/^#define HAVE_DECL_BASENAME 0$$/#define HAVE_DECL_BASENAME 1/' \
 		-e 's/^#define HAVE_DECL_SBRK 0$$/#define HAVE_DECL_SBRK 1/' \
 		-e 's/^#define HAVE_DECL_STRSIGNAL 0$$/#define HAVE_DECL_STRSIGNAL 1/' \
+		-e 's/^#define rlim_t long$$/\/\* #undef rlim_t \*\//' \
 		-e 's/^#define HAVE_DECL_STRSTR 0$$/#define HAVE_DECL_STRSTR 1/' \
 		auto-build.h > auto-build.h.tmp
 	mv auto-build.h.tmp auto-build.h
 .endif
 
 #
 # These are copies of files we build for the build host that are used
 # by the genfoo programs.
 #
 
